apps/server/src/modules/idp-console/idp-console.module.ts

```diff
@@ -0,0 +1,81 @@
+import { Module } from '@nestjs/common';
+import { ConsoleModule } from 'nestjs-console';
+import { CqrsModule } from '@nestjs/cqrs';
+import { ConsoleWriterModule } from '@infra/console';
+import { ConfigModule } from '@nestjs/config';
+import { HttpModule, HttpService } from '@nestjs/axios';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { Logger, LoggerModule } from '@src/core/logger';
+import { MikroOrmModule } from '@mikro-orm/nestjs';
+import { DB_PASSWORD, DB_URL, DB_USERNAME } from '@src/config';
+import { SchulconnexRestClient } from '@infra/schulconnex-client';
+import { AccountModule } from '@modules/account';
+import { OauthAdapterService } from '@modules/oauth';
+import { defaultMikroOrmOptions } from '@modules/server';
+import { ALL_ENTITIES } from '@shared/domain/entity';
+import { UserService } from '@modules/user';
+import { UserRepo } from '@shared/repo';
+import { UserDORepo } from '@shared/repo/user/user-do.repo';
+import { RoleModule } from '@modules/role';
+import { IdentityManagementModule } from '@infra/identity-management';
+import { RegistrationPinModule } from '@modules/registration-pin';
+import {
+	SynchronizationEntity,
+	SynchronizationRepo,
+	SynchronizationService,
+	SynchronizationUc,
+} from '@modules/synchronization';
+import { IdpSyncConsole } from './idp-sync-console';
+
+@Module({
+	imports: [
+		ConfigModule.forRoot({ isGlobal: true }),
+		ConsoleModule,
+		ConsoleWriterModule,
+		LoggerModule,
+		CqrsModule,
+		MikroOrmModule.forRoot({
+			...defaultMikroOrmOptions,
+			type: 'mongo',
+			clientUrl: DB_URL,
+			password: DB_PASSWORD,
+			user: DB_USERNAME,
+			allowGlobalContext: true,
+			entities: [...ALL_ENTITIES, SynchronizationEntity],
+			debug: true,
+		}),
+		IdentityManagementModule,
+		HttpModule,
+		AccountModule,
+		RegistrationPinModule,
+		RoleModule,
+	],
+	providers: [
+		SynchronizationRepo,
+		SynchronizationService,
+		SynchronizationUc,
+		OauthAdapterService,
```

Comment:
this is also from another module, import the module instead.

Reply:
If you meant Synchronization module then I've covered it in this comment: https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859#discussion_r1556015029 .

If you meant `OauthModule` then it's the same story as for the `UserModule` - importing it caused an app to require `FilesStorageProducer`:
```
Error: Nest can't resolve dependencies of the FilesStorageProducer (?, LegacyLogger, ConfigService). Please make sure that the argument AmqpConnection at index [0] is available in the FilesStorageClientModule context.
```

Reply:
I mean the OauthAdapterService. you need to import the OauthModule instead.

Reply:
Turned out the `OauthAdapterService` (and the whole `OauthModule`) is not needed anymore after additional changes regarding other imports.